lib/Artwork.php

```diff
@@ -195,14 +195,29 @@ public static function Get(?int $artworkId): Artwork{
 		return $result[0];
 	}
 
+	public static function GetByUrlPath($artistUrlName, $artworkUrlName): Artwork{
```

Comment:
I think we may want to think about how to ensure that this property is unique for each artwork. At the moment it's possible to have overlaps that make it impossible to access an artwork - or create links for one artwork that take you to a different one.

Here are some steps that reproduce this behaviour;

1. Submit artwork with artist name 'test-artist' & artwork name 'test-artwork'
2. Submit artwork with same artist/artwork name & a different image
3. In the artwork admin queue, decline the first entry and approve the second
4. In the artworks list, the second (approved) item will be present, but clicking on it will take you to the page for the first (declined) item.

It's a contrived example but I think this scenario could happen fairly easily if the first time an artwork is submitted it's rejected for missing proof of PD or incorrect metadata.

As for how to fix it, I don't have any perfect ideas at the moment. One way would be to check that there are no `approved` artworks with the same URL before allowing approval, and have this method filter by status as well.

n.b. I don't think this discussion needs to block this PR - I think the problem would have been present in the version using the cache as well.

Reply:
Thanks for noticing that. Let's address it in a follow-up PR. Your comment also reminded me to add a small commit (6d6258a) to this function that limits the query to Approved and In Use artwork. (Unverified and declined artwork can still be viewed on the admin page.) Despite that change, there's still a possibility of submitting two artworks with the same name and artist, which we will fix.